.github/workflows/suggest-review-fixes.yml

```diff
@@ -0,0 +1,112 @@
+# Do not edit this file directly. Instead, edit the example and run:
+# npm run gen-workflows
+
+name: Suggest Review Fixes
+on:
+  workflow_call:
+    inputs:
+      profile:
+        description: Optional Warp Agent profile name to use for Warp Agent.
+        required: false
+        type: string
+        default: ''
+    secrets:
+      WARP_API_KEY:
+        description: Warp API key used by the Warp Agent.
+        required: true
+jobs:
+  suggest_review_fixes:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Fetch review comments
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const { owner, repo } = context.repo;
+            const reviewId = context.payload.review.id;
+            const prNumber = context.payload.pull_request.number;
+
+            core.info(`Fetching comments for review ${reviewId} on PR #${prNumber}`);
+
+            const { data: comments } = await github.rest.pulls.listCommentsForReview({
+              owner, repo,
+              pull_number: prNumber,
+              review_id: reviewId
+            });
+
+            core.info(`Found ${comments.length} comments.`);
+            fs.writeFileSync( 'review_comments.json', JSON.stringify(comments, ['id', 'path', 'line', 'start_line', 'body', 'diff_hunk'], 2));
+      - name: Construct Prompt
+        id: prompt
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const prompt = `You have a list of review comments in review_comments.json. For each comment:
+            1. Determine if it's a small, straightforward fix (typos, naming, simple refactors, style issues).
+            2. If yes: propose a code suggestion (DO NOT modify files directly).
+            3. If no (too complex, requires design decisions, unclear): skip it.
+
+            Output a file responses.json with this format:
+            {
+              "<comment_id>": {
+                "reply": "Your minimal explanation here, followed by a suggestion block",
+              }
+            }
+
+            When you can propose a specific code fix, the \`reply\` field should contain a suggestion block in your comment body. 
+            -  Format:
+            \`\`\`suggestion
+            <only the code you are proposing as the replacement for the selected lines>
+            \`\`\`
+            -   The suggestion block must contain **only the changed code**, with no extra, unchanged context lines before or after the change.
+            -   For suggestions where the original code spans multiple lines, you must also include a \`start_line\` field in the JSON comment representing the first line of the range covered by the suggestion; \`line\` should be the last line of the range.
+
+            Only include comments you can address. Omit comments you skipped.
```

Comment:
One thing that should work eventually (but currently doesn't b/c of permissions issues) is having the agent itself validate the JSON before finishing. I've found it often misformats it in one way or another.